homeassistant/components/mqtt/mixins.py

```diff
@@ -260,8 +265,42 @@ async def __call__(
         """Define setup_entities type."""
 
 
+async def async_discover_platform(
+    hass: HomeAssistant, platform_domain: str, schema: vol.Schema
+) -> None:
+    """Set up platform if there is manual config."""
+
+    async def _async_discover_platform(event: Event | None) -> None:
```

Comment:
May be you can add a doc string to clarify when this is called and what happens then.